unsafe/src/main/java/org/apache/spark/unsafe/types/Interval.java

```diff
@@ -30,6 +32,42 @@
   public static final long MICROS_PER_DAY = MICROS_PER_HOUR * 24;
   public static final long MICROS_PER_WEEK = MICROS_PER_DAY * 7;
 
+  private static String unitRegex(String unit) {
+    return "(?:\\s+(-?\\d+)\\s+" + unit + "s?)?";
```

Comment:
add more comments explaining the regex, since most people would need to look up documentation on the regex.